src/data/equiv/mul_add.lean

```diff
@@ -45,6 +45,14 @@ def mul_hom.inverse [has_mul M] [has_mul N] (f : mul_hom M N) (g : N → M)
                ... = g (f (g x * g y)) : by rw f.map_mul
                ... = g x * g y : h₁ _, }
 
+/-- The inverse of a bijective `monoid_hom` is a `monoid_hom`. -/
+@[to_additive "The inverse of a bijective `add_monoid_hom` is an `add_monoid_hom`."]
+def monoid_hom.inverse {A B : Type*} [monoid A] [monoid B] (f : A →* B) (g : B → A)
+  (h₁ : function.left_inverse g f) (h₂ : function.right_inverse g f) :
+  B →* A :=
+{ map_one' := show g 1 = 1, by rw [← f.map_one, h₁],
+  .. (f : mul_hom A B).inverse g h₁ h₂, }
+
```

Comment:
In the interest of reducing build times, it might make sense to spin this change and the other new `inverse`s off into their own PR.

Reply:
I actually span out the new `non_unital_alg_hom` so we could reuse the oleans from this PR for the other changes.